faf-java-server-app/src/main/java/com/faforever/server/security/BanDetails.java

```diff
@@ -25,27 +27,38 @@
   @Id
   @Column(name = "id")
   private Integer id;
-
   @ManyToOne
   @JoinColumn(name = "player_id")
+  @NotNull
   private User user;
-
   @ManyToOne
   @JoinColumn(name = "author_id")
   private User author;
-
   @Column(name = "reason")
+  @NotNull
   private String reason;
-
   @Column(name = "expires_at")
   private Timestamp expiresAt;
-
   @Column(name = "level")
   @Enumerated(EnumType.STRING)
   private BanScope scope;
+  @Column(name = "revoke_reason")
+  private String revokeReason;
+  @ManyToOne
+  @JoinColumn(name = "revoke_author_id")
+  private User revokeAuthor;
+  @Column(name = "revoke_time")
+  private OffsetDateTime revokeTime;
 
-  @Formula(value = "(SELECT count(1) FROM ban_revoke WHERE ban_revoke.ban_id = id)")
-  private boolean revoked;
+  public boolean isActive() {
+    if (revokeTime != null && revokeTime.isBefore(OffsetDateTime.now())) {
+      return false;
+    }
+    if (expiresAt == null) {
+      return true;
+    }
+    return expiresAt.after(Timestamp.from(Instant.now()));
+  }
```

Comment:
That's due to some auto-formatting stuff on the latest IntelliJ version after commiting. It gives no warning, but also does not format the whole file. Very annoying.
Maybe we need to checkin a code-style.

Reply:
The code-style *is* checked in (see https://github.com/FAForever/faf-java-server/tree/develop/.idea/codeStyles). But auto formatting does not add or remove if-statements.

Regardless, I see now that it can not be shortened and is therefore fine as it is.